src/Paket.Core/PaketConfigFiles/InstallModel.fs

```diff
@@ -784,7 +784,13 @@ module InstallModel =
 
     let filterReferences (references:string Set) (this:InstallModel) =
         this
-        |> mapCompileLibReferences (Set.filter (fun reference -> Set.contains reference.Name references |> not))
+        // HACK: workaround for https://github.com/fsprojects/Paket/issues/2811
+        //  * DO remove FW-References which are already referenced by the user
+        //  * DO NOT remove package references, where the user already has a reference
+        // Example where this is relevant:
+        // 1) Pre-existing reference to the FW-Assembly System.IO.Compression
+        // 2) user adds nuget package https://www.nuget.org/packages/System.IO.Compression/
+        //|> mapCompileLibReferences (Set.filter (fun reference -> Set.contains reference.Name references |> not))
```

Comment:
I'm not sure this is the place to do it. InstallModel is / or should be independent from msbuild/project file quirks

Reply:
``filterReferences`` is called from ProjectFile.fs:
![image](https://user-images.githubusercontent.com/4236651/37384703-9fdfad8a-2750-11e8-878e-c1bf6d286d58.png)

This is the only place that calls it.

Reply:
2 and 3 are fine by me. Basically it's less about where it is used but that the InstallModel-API speaks for itself and is consistent in its naming (it is already hard to understand as it is)